lightning/src/chain/keysinterface.rs

```diff
@@ -345,17 +333,17 @@ pub trait KeysInterface: Send + Sync {
 /// A simple implementation of ChannelKeys that just keeps the private keys in memory.
 pub struct InMemoryChannelKeys {
 	/// Private key of anchor tx
-	funding_key: SecretKey,
+	pub funding_key: SecretKey,
```

Comment:
I think even for this concrete implementation, it would be better to not make the private key properties public. Perhaps add a test config constraint?

Reply:
You need them to be able to sign transactions when handling SpendableOutput events, so there needs to be *some* way to get at them.

Reply:
Don't know the full context, but might it be better to expose a way to use them (e.g. expose a `sign_tx` method) rather than exposing the keys themselves?

Reply:
If the signing only handles in `impl for InMemoryChannelKeys`, they may not have to be pub, right?

Reply:
No, we dont require signing functions for SpendableOutputs events as those are purely external and can happen a month later. eg the use in the tests at https://github.com/rust-bitcoin/rust-lightning/blob/master/lightning/src/ln/functional_tests.rs#L4299

Reply:
show my volume with BTC in the wallet